sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_administration_client.hpp

```diff
@@ -319,4 +290,45 @@ namespace Azure { namespace Security { namespace Attestation {
         Azure::Core::Context const& context = Azure::Core::Context{}) const;
   };
 
+  /** @brief Construct a new AttestationAdministrationClient object.
+   *
+   * The AttestationAdministrationClientFactory class is a factory class for instantiating new
+   * AttestationAdministrationClient objects.
+   *
+   */
+  class AttestationAdministrationClientFactory final {
+  public:
+    /**
+     * @brief Construct a new Attestation Administration Client object.
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param credential The authentication token to use.
+     * @param options The options to customize the client behavior.
+     * @return std::unique_ptr<AttestationAdministrationClient> The newly created client.
+     */
+    static AttestationAdministrationClient Create(
+        std::string const& endpoint,
+        std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+        AttestationAdministrationClientOptions const& options
+        = AttestationAdministrationClientOptions(),
+        Azure::Core::Context const& context = Azure::Core::Context{});
+    /**
+     * @brief Construct a pointer to a new Attestation Administration Client object.
+     *
+     * @note It is the responsibility of the caller to manage the lifetime of the returned
+     * AttestationAdministrationClient object, typically by constructing a std::unique_ptr or
+     * std::shared_ptr from this pointer.
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param credential The authentication token to use.
+     * @param options The options to customize the client behavior.
+     */
+    static std::unique_ptr<AttestationAdministrationClient> CreatePointer(
```

Comment:
Can we have just one factory method - for example, with the `unique_ptr` (more flexible), and maybe add another if there's a request? Unless it is very common to worry about dynamic memory fragmentation when creating and destroying hordes of clients, I think it is more confusing to discover both methods in IntelliSense, and then be thinking about benefits of each? `unique_ptr` privides RAII, the same as byvalue return, and ownership can be transferred to other smart pointers. I'd name it `Create()`, and make it return `unique_ptr`.

Reply:
Every service client in the Azure SDK allows you to create a client as a value type. If we only allowed the pointer version of `Create`, the attestation service would be unique because it did not allow you to create a local service client.

I thought that was too great a loss of functionality to remove the ability to create a local client.

And I *hate* inconsistency.

Reply:
`unique_ptr` is local, you can have it in function scope and RAII will behave the same. Most of the time, no one would think too much about comparing local value vs local smart pointer (unless it is perf-critical or you don't want to fragment dynamic memory), you just get something, and you work with it.
If other factories also have by-value returns, or even both, I'd recommend to also update them to only have one method returning smart pointer.

Reply:
The pattern I see THROUGHOUT the Azure SDK for C++ is:

```c++
auto containerClient
      = BlobContainerClient::CreateFromConnectionString(GetConnectionString(), containerName);

  containerClient.CreateIfNotExists();
```

Literally every single sample I could find follows this pattern.


If Create returned a `std::unique_ptr`, that semantic would not be possible. You would ONLY be able to refer to an `AttestationClient` by reference (`std::unique_ptr` is NOT a value type - it is a container for a reference type).

Unfortunately the ship has sailed w.r.t. the pattern of Azure C++ SDK clients returning clients by value (see all the storage APIs which have a `CreateFromConnectionString` factory method. Every one of those returns a value type, none of them return pointers.

Reply:
Why do we need both `Create()` and `CreatePointer()`? Can we just have `Create()` that returns the client as a value-type, or is there some strong customer need for `CreatePointer()` (maybe specific to attestation)?

Using that is easier and simpler, like the `CreateFromConnectionString()` sample snippet you should above from storage.